wgsl/diagrams/operators_1.mmd

```diff
@@ -0,0 +1,61 @@
+flowchart TD
+    classDef ltrGraph fill:#797;
+    classDef rtlGraph fill:#779,stroke-width:4px;
+    classDef noneGraph fill:#977,stroke-width:4px,stroke-dasharray: 5 5;
+    subgraph legend["Legend"]
+    legend0["Left-to-right"]:::ltrGraph
+    legend1("Right-to-left"):::rtlGraph
+    legend2["Requires parentheses"]:::noneGraph
+    end
+    A["(...)"]
+    subgraph primary[" "]
+    primary0["a()"]
+    primary1["a[]"]
+    primary2["a.b"]
+    end
+    subgraph unary[" "]
+    unary0("-a")
+    unary1("!a")
+    unary2("~a")
+    unary3("*a")
+    unary4("&a")
+    end
+    subgraph multiplicative[" "]
+    multiplicative0["a*b"]
+    multiplicative1["a/b"]
+    multiplicative2["a%b"]
+    end
+    subgraph additive[" "]
+    additive0["a+b"]
+    additive1["a-b"]
+    end
+    subgraph shift[" "]
+    shift0["a&lt;&lt;b"]
+    shift1["a>>b"]
+    end
+    subgraph relational[" "]
+    relational0["a&lt;b"]
+    relational1["a>b"]
+    relational2["a&lt;=b"]
+    relational3["a>=b"]
+    relational4["a==b"]
+    relational5["a!=b"]
+    end
+    class primary ltrGraph;
+    class unary rtlGraph;
+    class multiplicative ltrGraph;
+    class additive ltrGraph;
+    class shift noneGraph;
+    class relational noneGraph;
+    A-->primary
+    primary-->unary
+    unary-->multiplicative
+    unary-->shift
+    unary-->binary0["a&b"]:::ltrGraph;
+    unary-->binary1["a|b"]:::ltrGraph;
+    unary-->binary2["a^b"]:::ltrGraph;
+    multiplicative-->additive
+    additive-->relational
+    shift-->relational
+    relational-->shortcircuit0["a&&b"]:::ltrGraph;
+    relational-->shortcircuit1["a||b"]:::ltrGraph;
```

Comment:
It would be very nice if this chart could have the same vertical ordering as the table form of the same thing. I made [some](https://mermaid.live/edit#pako:eNqFVl1P2zAU_SvGEwhY07UFVJEKJCSmvewJ9kaY5CROY-E6ne3QVcB--67t5sNJWipVss4599P2dd5wUqQUhzjjxSbJidTo130kEPwSTpS6pxniWv6QZJ2jjHEefplfzxcdhdTcU8yvR0rL4oUGG5bqPLxc_-2aiELQts31fN6zqYCUKMhMkm2IrtBV1xMTr0yxmFPnSUsi1JpIKvTOPpQ0XaDjY5QytebgxcR2TlQZL20SnC6pSJ8i_NMuIvzsBA6fWCLTgS4CyZa5Bj4Mw6oxben0NMIPRmK0HGwifAbSqkNt6Qy8PtA_JZNUIZdxThVVznndIGdikrKLO7A6HY_HZ3WOdRFryVZEbkGAanKHmQrI6VkXnhr46bkLm8zIOK7hOngdqhTdQBaZQPUBgZJbmOnIUQebAfavg10Adt7BLgE7abB-HquSa7bmLCGavVIvIZ-yDTiP97C2D9_2sbYdx4faQdKU9RKoQBv6a9zDbdDgkFuVs0x7Pi1iHZ5wvTD_2Oes09vbQ14l5VBWIQj3XDdw7T8eIF2AIWZWmd0MsRfWbpC6NNTNIHVlqKObQ-XErHcWHeTKiP27umgrbC3vhxS2pt_DiqH9KqROmExK1t22hnBp7cmrLXTZvX8S3M7B6uKijsqR9irVQ9qj_IM-bF6d12HWnrpmnHtcs497BK7JzQjveG560ZPAOP8uskImFBUypZKJJSrMs5LCMN0wnbfeBc7Ei-qerCC43UUHV0aBJv71BIGtrX0cLNbKqjKd7uY6rB_1FkJORtPRDH36_gB4l-gS-gPvW7FRkDSCRwAtm8F_BzF3u-vNaIDtvramJUD-fvpcq5oK2l2UIXA6BM6GxiOwVc96HWz63ZpRA7i3L95t-YSf4hFeUbkiLIWvmDejjjC0cEUjHMIyJfIlwpH4AB0pdfG4FQmGj4SSjnC5Tomm94xAu1c4zAhX9OM_7jn85A) hacky [attempts](https://mermaid.live/edit#pako:eNqFVl1P2zAU_SvGExWwpqMFVJEKJCSmvewJ9kaY5CROY-E6me3QVcB--66d5sOpW5CQonPO_fDxzU3fcFKkFIc448U6yYnU6Nd9JBD8JZwodU8zxLX8IUmZo4xxHn6ZX88XA4XU3FHMr8dKy-KFBmuW6jy8LP8OQ0QhaD_mej7fiWmAlCjoTJJNiK7Q1TATE69MsZjTOpOWRKiSSCr0Nj6UNF2g42OUMlVyyGJq10lUFS9tE5wuqUifIvzTPkT4uRbU-LklMh3oIpBsmWvgwzBsjOlLpycRfjASo-UQE-FTkDYO9aUzyPpA_1RMUoXqjnOqqKqTtwbVIaYp-3AHUSeTyeS07bE9RCnZisgNCFBLbjFzAnJyOoSnBn56HsKmMzKJW7gt3paqxLCQRc7h9AGBI_cw48jRAJsB9m-AXQB2NsAuARt12G4fq4prVnKWEM1eqdOQS1kDzuI9rPXh2z7W2nF8yA6SpmyngQa0pb_GO7gtGhxKq3KWaSenRWzCEdcL8x-7nE16e3soq6QcjlUIwp3UHdzmjz1kXcDHzJqwGx97YeO81KWhbrzUlaGObjzHiZlo5noUu-_joq-w_b4fUti-f_sVKi-kTphMKra1ffS5sC65p6bdW82LhrykHf12qTqUO5j-8Ga-_Kydkm79Olzn-x5BbVi3cgeZOwt2JLB-v4uskAlFhUypZGKJCvMZSGH5rZnOe3ucM_GihpMQBLfbK3ffIsDtkZypaMVTZxRaeObcv03Ru2fP1Q8k27Sm0Ue9gZbPx9PxDH36vQHwLtEV-Avfs2Kt4NAIlj5adov-Dkptp8PZyQDbuehtR4DceXC5ni0N5DjoglMfOPOtQ2Ab83euoruv3k7y4M69eszfy0_xGK-oXBGWwq-WN6OOMFi4ohEO4TEl8iXCkfgAHal08bgRCYYfBRUd46pMiab3jIDdKxxmhCv68R9Fcvms), but wasn't very successful. The one [promising](https://mermaid.live/edit#pako:eNqFVl1P2zAU_SvGE4iypKOlVUUqkJCY9rIn2BthkpM4jYWbdLZDVwH77bu2mw8naUFCis459-vkxu4bjouE4gCnvNjGGREK_boPcwR_MSdS3tMUcSV-CLLJUMo4D74srhfLjkIo7igW155Uonih_pYlKgtmm7_dkLzIaTvmerHoxVRAQiR0JsguQHM072Zi-SuTLOLUZlKC5HJDBM3VPj4QNFmi01OUMLnhkEXXtklkGa1ME5yuaJ48hfineQjxsxVY_NIQqfJV4Qu2yhTwQRBUxrSlk_MQP2iJ1nKICfEIpJVDbekUsj7QPyUTVCLbcUYllTZ5bZAN0U2ZhzuIOh-Px6O6x3qIjWBrInYgQDW5x_QE5HzUhScafnruwrozMo5quC5elyrzbiGDXML0PoGRW5h25KSDTQH718GuALvoYDPAzhqs38e65IptOIuJYq_UaciljAEX0QHW-PDtEGvsOD1mB0kS1mugAk3pr1EPN0X9Y2llxlLl5DSISXjG1VL_Ry5nkt7eHssqKIexipxwJ3UD1_mjAdIWGGKmVdjNEHtl4gapmaZuBqm5pk5ujo0Tsd4uWsiOEbnf6rKtMLO8H1OYmX4PK4beVyFUzERcsu5rawjb1oG-2kLb3fsnxc05WH24qKOypPmU6kPaodxFHw6v9nWYNVvXHOcO17zHAwJrcnOEdzI3XvQkcJx_z9NCxBQVIqGC5StU6GslgcN0y1TWuhc4y19kd7N8_9ZWdz9KgM1E7SUwWNOLs2N1mqmzNjU8GViATr69RHf5qHbQ76U38abelTfz5ujTKwzAu1iVYDFckcVWwtwI7hG0au6OOyi4XxDnmAe4bBwo96O6K-FyLWsqqG2iGwpk5WrP4-Y9tM6uAdx5X23TsIfXVKwJS-DHy5sWhxjGXtMQB_CYEPES4jD_AB0pVfG4y2MMvw1K6uFykxBF7xkBi9Y4SAmX9OM_YTP6yQ) direction was to make a 4th subgraph type and use it for the binary and shortcircuit operators. The result still sucks though.

Ultimately this may be a diagram where we have to hand-draw or hand-modify it to look nice - or maybe try a different diagram generator that provides more control over positioning. But we don't necessarily have to do that right now; the chart you have is perfectly serviceable. Maybe we could leave a TODO in the mermaid file to try to make it match the table ordering.

Reply:
@kainino0x I think it should be fine to have graph (and order of expressions) deviate a bit from the table, as table is a bit more constrained compared to these two. I'd like to bring this suggestion to the attention of @dneto0 and @alan-baker as these attempts have very their strengths too.

I agree with current diagram generator's behavior being a bit restrictive in expression, especially with arrows (where diagrams end up having a lot of overlaps even for very simple scenarios sometimes). But I based the current one on your comment back then and I think it actually represents the idea quite well, two anchor resources I was referring while tabularizing were your graph & the new syntax styling in the open PR (where it is easier on eye to see symbols and stuff inlined).

Thank you!

Reply:
(Though, to be honest, I did not consider not-using Mermaid. If that is acceptable for other editors, I actually think of drafting some compaction/balance of latest commit and ideas captured by @kainino0x above)

Reply:
I think this diagram is fine, and there's no real need to redo it in another tool / draw it by hand. The editability of Mermaid is probably worth having a bit less control over the diagram.

Reply:
@kainino0x Agreed, thank you very much for your guidance and review to make it more tolerable!